src/main/java/sc/fiji/bdvpg/scijava/processors/BdvHandlePostprocessor.java

```diff
@@ -0,0 +1,61 @@
+package sc.fiji.bdvpg.scijava.processors;
+
+import bdv.util.BdvHandle;
+import org.scijava.module.Module;
+import org.scijava.module.process.AbstractPostprocessorPlugin;
+import org.scijava.module.process.PostprocessorPlugin;
+import org.scijava.object.ObjectService;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import sc.fiji.bdvpg.scijava.BdvHandleHelper;
+import sc.fiji.bdvpg.scijava.services.BdvSourceDisplayService;
+import sc.fiji.bdvpg.scijava.services.GuavaWeakCacheService;
+
+import java.util.function.Consumer;
+
+/**
+ * Ensures BdvHandle is stored into ObjectService
+ * and all containing Sources as well are stored into the BdvSourceDisplayService and
+ * BdvSourceService
+ * Also fix Bdv Close operation
+ */
+
+@Plugin(type = PostprocessorPlugin.class)
+public class BdvHandlePostprocessor extends AbstractPostprocessorPlugin {
+
+    @Parameter
+    BdvSourceDisplayService bsds;
+
+    @Parameter
+    ObjectService os;
+
+    @Parameter
+    GuavaWeakCacheService cacheService;
+
+    public static Consumer<String> log = (str) -> System.out.println(BdvHandlePostprocessor.class.getSimpleName()+":"+str);
+
+    @Override
+    public void process(Module module) {
+
+        module.getOutputs().forEach((name, object)-> {
+            if (object instanceof BdvHandle) {
+                BdvHandle bdvh = (BdvHandle) object;
+                log.accept("BdvHandle found.");
+                //------------ Register BdvHandle in ObjectService
+                os.addObject(bdvh);
```

Comment:
Instead of doing calling `ObjectService#addObject()`, you could also publish an [`ObjectCreatedEvent`](https://javadoc.scijava.org/SciJava/index.html?org/scijava/object/event/ObjectCreatedEvent.html), which leads to automatic registration with the `ObjectService`.

Then `BdvSourceService` and `BdvSourceDisplayService` could just as well listen to these `ObjectEvent`s and react accordingly.

I'm unsure what would be the best strategy. Maybe @ctrueden has some better advice.

Reply:
Ok, I don't remember how to use the `EventService`. Do you have a simple example somewhere ?

Reply:
This might be of use:

https://github.com/scijava/scijava-common/blob/560ad964a7a1e859edd4dbe4fc48d7d9e09523f2/src/main/java/org/scijava/display/DefaultDisplayService.java#L211-L218